New branche

Spolpinka opened the pull request:
4 теста по созданию User

Spolpinka committed:
первые 3 теста

Spolpinka committed:
первые 3 теста
+ 4ый тест

Spolpinka committed:
первые 3 теста
+ 4ый тест

Spolpinka committed:
первые 3 теста
+ 4ый тест

Spolpinka committed:
первые 3 теста
+ 4ый тест
+ параметризированный тест

alekseykoshelev commented on src/test/java/pro/sky/testjunit/tests/UserTest.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Test
+    public void defaultConstructorUserTest() {
+        assertNotNull(emptyUser);
```
Тут еще можно проверить, что поля null.

alekseykoshelev commented on src/test/java/pro/sky/testjunit/tests/UserTest.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Test
+    public void isEmailCorrectTest() {
+        assertTrue(first.getEmail().contains("@") && first.getEmail().contains("."));
```
Проверки правильные, но тут ты проверяешь, что созданный тобой пользователь сделан правильно. Но ничто не мешает мне в коде создать пользователя с таким вот email `foobar`. 
То есть сами проверки нужно сделать в конструкторе, чтобы исключить возможность создания неправильного объекта. Можно выбрасывать исключения, если передаются неверные значения. 
В тесте же нужно передавать неправильные значения и проверять, что выбрасывается исключение. Это можно сделать через assertThrows()

alekseykoshelev commented on src/test/java/pro/sky/testjunit/tests/UserTest.java:
```diff
@@ -0,0 +1,51 @@
+
+public class UserTest {
+
+    User first;
```
Как мне кажется, можно не делать отдельный класс для констант и вот эти поля можно сделать локальными переменными в каждом тесте. Они ведь так и используются по сути.

Spolpinka committed:
первые 3 теста
+ 4ый тест
+ параметризированный тест
+ выброс исключения в конструкторе + его проверка в тесте

Spolpinka committed:
первые 3 теста
+ 4ый тест
+ параметризированный тест
+ выброс исключения в конструкторе + его проверка в тесте
